cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -54,23 +54,23 @@ type SpanHandlerBuilder interface {
 }
 
 // NewSpanHandlerBuilder returns a span handler
-func NewSpanHandlerBuilder(opts ...basicB.Option) (SpanHandlerBuilder, error) {
+func NewSpanHandlerBuilder(cOps *CollectorOptions, flagOpts *flags.Flags, opts ...basicB.Option) (SpanHandlerBuilder, error) {
```

Comment:
Yeah, almost. I left one comment about unnecessary private struct, but the other thing I was suggesting was instead of this:
```go
cOpts := new(CollectorOpts).InitFromViper(v)
b := NewSpanHandlerBuilder(cOpts, ...)
b.BuildHandlers()
```

merge CollectorOpts into SpanHandlerBuilder

```go
b := new(SpanHandlerBuilder).InitFromViper(v)
h1, h2 := b.BuildHandlers(basicOpts...)
```

Basically I don't see the need why builder and options need to be separate struct - they are a single struct in AgentBuilder.